src/compiler/types.ts

```diff
@@ -1064,6 +1064,7 @@ module ts {
         getGlobalDiagnostics(): Diagnostic[];
         getSemanticDiagnostics(sourceFile?: SourceFile): Diagnostic[];
         getDeclarationDiagnostics(sourceFile?: SourceFile): Diagnostic[];
+        /* @internal */ getCompilerOptionsDiagnostics(): Diagnostic[];

```

Comment:
this is bad.  We now have multiple ways to get the same diagnostics.  Previously, diagnostics were entirely split, and any given diagnostic would only be in one of the four "getXXXDiagnostics" methods.  Now both getGlobalDiagnostics and getCompilerOptions diagnostics will return the same errors.  This makes it much harder for clients to know which one they should call.  If you do want to expose getCompilerOptionsDiagnostics, then i would remove the compiler options diagnostics from getGlobalDiagnostics.